client/packages/invoices/src/OutboundShipment/DetailView/OutboundLineEdit/allocation/useAllocationContext.ts

```diff
@@ -131,23 +130,46 @@ export const useAllocationContext = create<AllocationContext>((set, get) => ({
       isDirty: false,
       draftLines: [],
       nonAllocatableLines: [],
-      placeholderQuantity: null,
+      placeholderUnits: null,
       item: null,
-      allocateIn: AllocateIn.Units,
+      allocateIn: { type: AllocateInType.Units },
+      availablePackSizes: [],
       alerts: [],
     })),
 
+  setAllocateIn: (allocateIn: AllocateInOption) => {
+    const { draftLines } = get();
+
+    // Clear the existing allocated stock if changing how we are allocating
+    // This might be annoying... but should validate before spending time on
+    // reallocating on change of allocation type/pack size
+    const result = allocateQuantities(draftLines, 0, { allocateIn });
```

Comment:
Maybe we should move the existing allocated quantity to the placeholder value or something??